devito/compiler.py

```diff
@@ -379,7 +378,10 @@ def jit_compile(soname, code, compiler):
             with open(src_file, 'r') as f:
                 code = f.read()
             # Bypass the devito JIT cache
-            cache_dir = mkdtemp()
+            # Note: can't simply use Python's `mkdtemp()` as, with MPI, different
```

Comment:
But isn't it what you want. I would assume that in some case (like in the could) each mpi rank won't have access to a common tmp dir but each to their own and will each need to compile it.

Reply:
unfortunately not. Think about what's going on here. The user has _manually_ modified some C file, but of course he's done that only on a single rank. All MPI processes should eventually pick up that specific file. So this actually requires some further coordination in the case of multi-node execution. Single node MPI should be fine. 

Also, with MPI, the executable ran by the MPI processes must be *exactly the same*.